### PR TITLE
adding matrix multiplication

### LINKED ISSUE #34: Matrix addition , multiplication and transpose
## 🚀 Feature

Hi, I am a GSSoC participant I would like to implement this in java, python ,c, c++, and javascript

### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#reporting-new-issues)?

yes

## Pitch

implement matrix addition multiplication and transpose of a matrix

### PR DESCRIPTION
**Pull request by aaadddiii**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

yes

### Description

fixes #34 

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

issue #34

### CODE REVIEW

**Comment by aaadddiii** [comment]
@HarshCasper  can you please review my pr

**Review by iamrajiv — APPROVED** [review]

File: docs/mathematics/matrix-multiplication.mdx [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+## Algorithm
+
+- Input matrix `A` and matrix `B`.
+- Input `matrix A` 
+- Input `matrix B` 
+- loop from `i` = 0 to number of rows of the matrix `A`
+- loop from `j` = 0 to number of columns of the matrix `B`
+- loop from `k` = 0 to number of rows of the matrix `B`
+- `result[i][j]` += `matrix 1[i][k]` * `mattrix 2[k][j]`
+- print `result`
```
Comment (HarshCasper) [inline docs/mathematics/matrix-multiplication.mdx:26]: Use text for algorithm steps. Use formulae to complement them

File: docs/mathematics/matrix-multiplication.mdx [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+
+## Introduction
+
+In mathematics, matrix multiplication is a binary operation that produces a matrix from two matrices. For matrix multiplication, the number of columns in the first matrix must be equal to the number of rows in the second matrix.
+
+## Explanation
+
+The matrix after matrix multiplication is known as the matrix product, It has the number of rows of the first and the number of columns of the second matrix. The product of matrices `A` and `B` is denoted as `AB`
+
+For Example
```
Comment (HarshCasper) [inline docs/mathematics/matrix-multiplication.mdx:14]: `Let's have an example:`

File: docs/mathematics/matrix-multiplication.mdx [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+```
+
+## Complexity Analysis
+
+- Time Complexity: O(`r1` * `c2` * `c1`), where `r1` is the number of rows of first matrix and `c2` is the number of columns of second matrix and `c1` is the number of columns of first matrix   
+- Space Complexity: O(`r1` * `c2`)
+
+## Credits
+
+-   [Hari Krishnan U](https://github.com/Harikrishnan6336) for [Source Code in C++](https://github.com/TesseractCoding/NeoAlgo/blob/master/C-Plus-Plus/math/matrix_multiplication.cpp)
```
Comment (HarshCasper) [inline docs/mathematics/matrix-multiplication.mdx:170]: `for the implementation in C++`

File: docs/mathematics/matrix-multiplication.mdx [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+-   [Adithya Kurien Peter](https://github.com/aaadddiii) for [Source Code in Python](https://github.com/TesseractCoding/NeoAlgo/blob/master/Python/math/Matrix_multiplication.py)
```
Comment (HarshCasper) [inline docs/mathematics/matrix-multiplication.mdx:171]: `for the implementation in Python`

File: docs/mathematics/matrix-multiplication.mdx [hunk 5]
```diff
@@ -0,0 +1,171 @@
...
+for i in range(Row_1):
+    for j in range(Col_2):
+        for k in range(Col_1):
+            result[i][j] += matrix_1[i][k] * matrix_2[k][j]
+
+print("The resultant matrix is:")
+for i in range(Row_1):
+    for j in range(Col_2):
+        print(result[i][j], end=" ")
+    print()
```
Comment (HarshCasper) [inline docs/mathematics/matrix-multiplication.mdx:130]: Keep all of this in a function and use a driver code to run this

**Review by HarshCasper — CHANGES_REQUESTED** [review]

**Comment by aaadddiii** [comment]
@HarshCasper  I have made the changes please review it

**Review by HarshCasper — APPROVED** [review]

**Pull request merged**